dts/arm/nordic/nrf51822.dtsi

```diff
@@ -78,7 +79,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x40003000 0x1000>;
+			periph-idx = <0>;
```

Comment:
@mbolivar-nordic this is used by other vendors as well FYI.

Reply:
> @mbolivar-nordic this is used by other vendors as well FYI.

There are 2 cases where this is used by other vendors:

1. Where the peripheral index is actually a value that is going to get written into a peripheral register
2. Where the author is stuffing software configuration into devicetree

Case 1. was explicitly discussed and approved, including by @galak. Case 2. is just laziness and catering to HAL restrictions in devicetree, and was explicitly disapproved by @galak. I have done my best to forbid additional cases, but some have crept in because I can't review everything in enough detail to enforce this policy.

In the case of the Nordic drivers, I am unaware of any register where we need this value, and so unless I am missing something it is clearly case 2, which is forbidden. I think the fix belongs in the HAL, or we should stop using the HAL.

Reply:
> unless I am missing something it is clearly case 2, which is forbidden.

I left this comment before beginning my review. Now that I got to the commit log, I have to NAK this approach on the grounds outlined above. I force other SoC vendors to play by these rules whenever we find people are trying; we should not make an exception for ourselves out of convenience here.